routing/dht/dht.go

```diff
@@ -173,7 +173,9 @@ func (dht *IpfsDHT) getValueOrPeers(ctx context.Context, p peer.ID,
 		err = dht.verifyRecordOnline(ctx, record)
 		if err != nil {
 			log.Info("Received invalid record! (discarded)")
-			return nil, nil, err
+			// still return a non-nil record to signify that we received
+			// a bad record from this peer
+			record = new(pb.Record)
```

Comment:
also, an invalid record should not necessarily be the same as an empty record, malicious peers who send invalid stuff should maybe not be talked to, etc. we may want to deal with them specially later (distinct from a valid but empty record)